src/pages/resultsView/survival/SurvivalUtil.tsx

```diff
@@ -82,6 +120,103 @@ export function getEstimates(patientSurvivals: PatientSurvival[]): number[] {
     return estimates;
 }
 
+export function parseSurvivalData(s: string) {
```

Comment:
we should get tests on all the new functions in this file.   also, keep an eye toward performance.  there are some optimizations that could be made.  but it's only worth making them IF these function are going to be run for every sample in result set.   if not, then it's negligible.    you would know better what the "n" is here.